From 2d93efed33432679d95b49ae62399a2efe30b94e Mon Sep 17 00:00:00 2001
From: Lin Fuzhen <fuzhen.lin@freescale.com>
Date: Tue, 14 May 2013 12:28:23 +0800
Subject: [PATCH 198/304] ENGR00262497 i.mx5x/i.mx6x:support Google AOA v2.0
 on i.MX platform

-align the PID with AOA2.0 defiend by Google and align with Nexus 7
-add usb audio accessory in audio policy

Signed-off-by: Lin Fuzhen <fuzhen.lin@freescale.com>
---
 imx5x/init.freescale.usb.rc   |  8 ++++----
 imx6/etc/init.usb.rc          |  8 ++++----
 imx6/imx6.mk                  |  1 +
 sabresd_6dq/audio_policy.conf | 16 ++++++++++++++++
 4 files changed, 25 insertions(+), 8 deletions(-)

diff --git a/imx5x/init.freescale.usb.rc b/imx5x/init.freescale.usb.rc
index 1aa2907..f2e6707 100644
--- a/imx5x/init.freescale.usb.rc
+++ b/imx5x/init.freescale.usb.rc
@@ -48,7 +48,7 @@ on property:sys.usb.config=rndis,adb
 on property:sys.usb.config=mtp
     write /sys/class/android_usb/android0/enable 0
     write /sys/class/android_usb/android0/idVendor 18d1
-    write /sys/class/android_usb/android0/idProduct 2d02
+    write /sys/class/android_usb/android0/idProduct 4e41
     write /sys/class/android_usb/android0/functions ${sys.usb.config}
     write /sys/class/android_usb/android0/enable 1
     setprop sys.usb.state ${sys.usb.config}
@@ -56,7 +56,7 @@ on property:sys.usb.config=mtp
 on property:sys.usb.config=mtp,adb
     write /sys/class/android_usb/android0/enable 0
     write /sys/class/android_usb/android0/idVendor 18d1
-    write /sys/class/android_usb/android0/idProduct d02
+    write /sys/class/android_usb/android0/idProduct 4e42
     write /sys/class/android_usb/android0/functions ${sys.usb.config}
     write /sys/class/android_usb/android0/enable 1
     start adbd
@@ -65,7 +65,7 @@ on property:sys.usb.config=mtp,adb
 on property:sys.usb.config=ptp
     write /sys/class/android_usb/android0/enable 0
     write /sys/class/android_usb/android0/idVendor 18d1
-    write /sys/class/android_usb/android0/idProduct 2d03
+    write /sys/class/android_usb/android0/idProduct 4e43
     write /sys/class/android_usb/android0/functions ${sys.usb.config}
     write /sys/class/android_usb/android0/enable 1
     setprop sys.usb.state ${sys.usb.config}
@@ -73,7 +73,7 @@ on property:sys.usb.config=ptp
 on property:sys.usb.config=ptp,adb
     write /sys/class/android_usb/android0/enable 0
     write /sys/class/android_usb/android0/idVendor 18d1
-    write /sys/class/android_usb/android0/idProduct d02
+    write /sys/class/android_usb/android0/idProduct 4e44
     write /sys/class/android_usb/android0/functions ${sys.usb.config}
     write /sys/class/android_usb/android0/enable 1
     start adbd
diff --git a/imx6/etc/init.usb.rc b/imx6/etc/init.usb.rc
index 4cc2080..0d7c29f 100644
--- a/imx6/etc/init.usb.rc
+++ b/imx6/etc/init.usb.rc
@@ -11,7 +11,7 @@ on property:sys.usb.config=rndis
 on property:sys.usb.config=mtp
     write /sys/class/android_usb/android0/enable 0
     write /sys/class/android_usb/android0/idVendor 18d1
-    write /sys/class/android_usb/android0/idProduct 2d02
+    write /sys/class/android_usb/android0/idProduct 4e41
     write /sys/class/android_usb/android0/functions ${sys.usb.config}
     write /sys/class/android_usb/android0/enable 1
     setprop sys.usb.state ${sys.usb.config}
@@ -19,7 +19,7 @@ on property:sys.usb.config=mtp
 on property:sys.usb.config=mtp,adb
     write /sys/class/android_usb/android0/enable 0
     write /sys/class/android_usb/android0/idVendor 18d1
-    write /sys/class/android_usb/android0/idProduct d02
+    write /sys/class/android_usb/android0/idProduct 4e42
     write /sys/class/android_usb/android0/functions ${sys.usb.config}
     write /sys/class/android_usb/android0/enable 1
     start adbd
@@ -28,7 +28,7 @@ on property:sys.usb.config=mtp,adb
 on property:sys.usb.config=ptp
     write /sys/class/android_usb/android0/enable 0
     write /sys/class/android_usb/android0/idVendor 18d1
-    write /sys/class/android_usb/android0/idProduct 2d03
+    write /sys/class/android_usb/android0/idProduct 4e43
     write /sys/class/android_usb/android0/functions ${sys.usb.config}
     write /sys/class/android_usb/android0/enable 1
     setprop sys.usb.state ${sys.usb.config}
@@ -36,7 +36,7 @@ on property:sys.usb.config=ptp
 on property:sys.usb.config=ptp,adb
     write /sys/class/android_usb/android0/enable 0
     write /sys/class/android_usb/android0/idVendor 18d1
-    write /sys/class/android_usb/android0/idProduct d02
+    write /sys/class/android_usb/android0/idProduct 4e44
     write /sys/class/android_usb/android0/functions ${sys.usb.config}
     write /sys/class/android_usb/android0/enable 1
     start adbd
diff --git a/imx6/imx6.mk b/imx6/imx6.mk
index 50c505e..14cdd78 100644
--- a/imx6/imx6.mk
+++ b/imx6/imx6.mk
@@ -67,6 +67,7 @@ PRODUCT_PACKAGES += \
 	audio_policy.conf			\
 	tinyplay				\
 	audio.a2dp.default			\
+	audio.usb.default			\
 	tinycap					\
 	tinymix					\
 	libsrec_jni				\
diff --git a/sabresd_6dq/audio_policy.conf b/sabresd_6dq/audio_policy.conf
index 2fca29c..c709621 100644
--- a/sabresd_6dq/audio_policy.conf
+++ b/sabresd_6dq/audio_policy.conf
@@ -74,4 +74,20 @@ audio_hw_modules {
       }
     }
   }
+ usb {
+    outputs {
+      usb_accessory {
+        sampling_rates 44100
+        channel_masks AUDIO_CHANNEL_OUT_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_OUT_USB_ACCESSORY
+      }
+      usb_device {
+        sampling_rates 44100
+        channel_masks AUDIO_CHANNEL_OUT_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_OUT_USB_DEVICE
+      }
+    }
+  }
 }
-- 
1.8.0

